File: salesforce-okta/src/main/java/com/jinloes/salesforce/model/SalesforceTokenResponse.java
package com.jinloes.salesforce.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Represents the OAuth 2.0 token response returned by the Salesforce token endpoint.
 *
 * @param accessToken the Bearer token to use in subsequent Salesforce API calls
 * @param instanceUrl the Salesforce instance base URL; all API requests must be rooted here
 * @param tokenType always "Bearer" for the JWT Bearer flow
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record SalesforceTokenResponse(
    @JsonProperty("access_token") String accessToken,
    @JsonProperty("instance_url") String instanceUrl,
    @JsonProperty("token_type") String tokenType) {}

File: salesforce-okta-csharp/Models/SalesforceTokenResponse.cs
using System.Text.Json.Serialization;

namespace salesforce_okta_csharp.Models;

public record SalesforceTokenResponse(
    [property: JsonPropertyName("access_token")] string AccessToken,
    [property: JsonPropertyName("instance_url")] string InstanceUrl,
    [property: JsonPropertyName("token_type")] string TokenType
);
